### ISSUE TITLE
auro-button: should be using `aria-disabled` instead of `disabled`

### ISSUE THREAD

**Issue by jason-capsule42** [AlaskaAirlines/auro-button#101]
## Describe the bug
`disabled` has poor a11y support with screen readers. Switching to `aria-disabled` will resolve that issue.

## Additional context
https://a11y-101.com/development/aria-disabled

**Comment by geoffrich** [comment]
I am not convinced this is a worthwhile change and think we need user research on this to make sure this behavior is desirable. I've only been able to find one source that suggests switching to aria-disabled that you linked above -- is there any other research or article supporting this change? Anecdotally, some a11y people on twitter [wouldn't recommend](https://twitter.com/yatil/status/1264953677846110209) switching out disabled for aria-disabled.

This change will also add additional complexity to the component because aria-disabled only communicates the disabled state to the accessibility tree, and [does not prevent the button from being interacted with](https://codepen.io/stevef/post/short-note-on-the-disabled-attribute#what-aria-disabled-does-4). If we switch to aria-disabled, we would also need to prevent click handlers from triggering, and there is potential for other subtle bugs to slip in.

As noted in a previous thread, screen readers are still able to find the disabled button using the virtual cursor. It is also discoverable when navigating via form element shortcuts in NVDA (F key). I'm also able to reach it and have it read out when navigating via swipe with iOS VoiceOver. It is only not discoverable when tabbing. Is this an actual problem for screen reader users?

**Comment by jason-capsule42** [comment]
@geoffrich I'm not attached to this change. If you feel confident that our current implementation fulfills the a11y needs I am comfortable deferring to your judgement there. 

I believe that using `aria-disabled` and then interrupting the click function in JS is the right approach from a purely `aria` concern. However, I also don't think that means it's the best approach in all scenarios. It's probably less risky when implementing a one-off button in project where you control every step of the button behavior than it is when creating a component where we have no control over the JS being called when triggering the click event.

Based on @geoffrich's feedback I am support abandoning this change and sticking without current `disabled` implementation.
